src/main/scala/cromwell/engine/backend/jes/JesBackendCall.scala

```diff
@@ -85,21 +87,21 @@ class JesBackendCall(val backend: JesBackend,
         val status = Try(handle.run.checkStatus(this, handle.previousStatus))
         status match {
           case Success(s: TerminalRunStatus) => backend.executionResult(s, handle)
-          case Success(s) => handle.copy(previousStatus = Option(s)) // Copy the current handle with updated previous status.
+          case Success(s) => handle.copy(previousStatus = Option(s)).future // Copy the current handle with updated previous status.
           case Failure(e: Exception) =>
             // Log exceptions and return the original handle to try again.
             logger.warn("Caught exception, retrying: " + e.getMessage, e)
-            handle
-          case Failure(e: Error) => throw e // JVM-ending calamity.
+            handle.future
+          case Failure(e: Error) => Future.failed(e) // JVM-ending calamity.
           case Failure(throwable) =>
-            // Someone has subclassed Throwable directly?  Beware the jackwagons, fail the execution/workflow.
```

Comment:
Spiritually it's yours, but I think I wrote it based on your PR feedback.  :smile: